### PR TITLE
Consistent CPU binding list for the frameworks and oneCCL

### PR DESCRIPTION
**Pull request by khossain4337**
## Description
Updated the CPU binding list in the oneCCL and TensorFlow document along with updated information from the `frameowrks/2025.0.0` updates. The goal is to provide a consistent set of CPU bindings across the three frameworks (PyTorch, TensorFLow, oneCCL) pages, which does not use cores 0 and 52.

The PyTorch page is written in a way which does not explicitly refer to CPU bindings except the multi-CCS case. That list does not include either 0 or 52. The nature of the usage is different from the "1 rank per tile" usage, therefore the bindings are different.

## Type of Change

- [x] Documentation content update (new page, formatting/typo changes, adding more info, etc.)
- [ ] Functionality bug fix
- [ ] New feature (`mkdocs` feature, `mkdocs-material` style changes, HTML/CSS/JS customization, developer or repo tool)

## Checklist

- [x] I have run `make serve` or `make build-docs` locally and verified that my changes render correctly
- [x] I have added at least one Label to this PR

### CODE REVIEW

**Comment by kaushikvelusamy** [comment]
These are my final recommendations. 

export CPU_BINDING1="list:4:9:14:19:20:25:56:61:66:71:74:79" # 12 ppn to 12 cores 	
export CPU_BINDING2="list:4-7:8-11:12-15:16-19:20-23:24-27:56-59:60-63:64-67:68-71:72-75:76-79" # 12 ppn with each ppn with total 4 cores 
export CCL_WORKER_AFFINITY=42,43,44,45,46,47,94,95,96,97,98,99
unset CCL_WORKER_AFFINITY: # Default it will pick up the last 24 cores even if you don't specify anything in the mpirun

**Comment by khossain4337** [comment]
Added the updated CPU binding list as per discussion with Kaushik. I believe now all the datascience pages have consistent CPU bindings. In the oneCCL page and the TensorFlow page, I have added a few lines with high level rationale behind our suggestions.

**Comment by felker** [comment]
@kaushikvelusamy does this look good to you now?
